sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/Scan.java

```diff
@@ -21,9 +21,8 @@
 import org.apache.spark.sql.sources.v2.reader.streaming.ContinuousStream;
 import org.apache.spark.sql.sources.v2.reader.streaming.MicroBatchStream;
 import org.apache.spark.sql.types.StructType;
-import org.apache.spark.sql.sources.v2.SupportsContinuousRead;
-import org.apache.spark.sql.sources.v2.SupportsMicroBatchRead;
 import org.apache.spark.sql.sources.v2.Table;
+import org.apache.spark.sql.sources.v2.TableCapability;
```

Comment:
Nit: this commit includes non-functional changes. These docs changes are good to have, but they are not related to streaming and there is no need for this file to change in this commit. I would prefer to revert this and include it in a docs-only PR.

Reply:
Same with the changes in `WriteBuilder`.

Reply:
it's related to streaming: https://github.com/apache/spark/pull/24129/files#diff-d111d7e2179b55465840c9a81ea004f2R76

The only unnecessary change is the doc for batch. I replace `BATCH_READ` with a java doc link, to be consistent with streaming.

Reply:
Javadoc changes are non-functional changes. If you want to correct Javadoc, limit updates to files that are already changing in the PR.

As it is, this is a non-functional change in an unrelated file. The commit message is going to indicate this commit is related to streaming, so this could easily cause conflicts. Please remove it.

Reply:
I don't agree with this. I add doc for streaming and change the doc for batch a little bit to match the streaming doc style. It's very annoying to separate minor change like this. I do consider this change as self-contained.

Reply:
I think there are too many documentation-only changes for this PR to be committed. Please separate them into a different PR. A docs-only commit will be easy to review and easy to cherry-pick based on the description.

Reply:
Did I miss something here? The docs changes are totally related to the current PR (the doc would refer to deleted interface).

Reply:
The docs changes that I was referring to were moved already, this was just a poorly placed comment.